app.js

```diff
@@ -1,117 +1,186 @@
 const express = require("express");
+const db = require("./databaseConnect");
 const app = express();
 const port = 3001;
+const jwt = require("jsonwebtoken");
+const authMiddleware = require("./authMiddleware");
+
+db.init();
 
 var hash = require("pbkdf2-password")();
 
 var cors = require("cors");
+const { json } = require("express");
 
 app.use(cors());
 
 app.use(express.urlencoded({ extended: true }));
 // This is required to handle urlencoded data
 app.use(express.json());
-// This to handle json data coming from requests mainly post
 
-var users = {
-    admin: { name: "admin" },
-};
+app.use(express.static("public"));
+
+// This to handle json data coming from requests mainly post
 
 // when you create a user, generate a salt
 // and hash the password ('foobar' is the pass here)
 
-hash({ password: "1234" }, function (err, pass, salt, hash) {
-    if (err) throw err;
-    // store the salt & hash in the "db"
-    users.admin.salt = salt;
-    users.admin.hash = hash;
+app.listen(port, () => {
+    console.log(`Example app listening on port ${port}`);
 });
 
 function authenticate(name, pass, fn) {
-    if (!module.parent) console.log("authenticating %s:%s", name, pass);
-    var user = users[name];
     // query the db for the given username
-    if (!user) return fn(null, null);
-    // apply the same algorithm to the POSTed password, applying
-    // the hash against the pass / salt, if there is a match we
-    // found the user
-    hash({ password: pass, salt: user.salt }, function (err, pass, salt, hash) {
-        if (err) return fn(err);
-        if (hash === user.hash) return fn(null, user);
-        fn(null, null);
-    });
+    db.getUserByUsername(name)
+        .then((res) => {
+            const user = res.rows[0];
+            if (!user) return fn(null, null);
+            // apply the same algorithm to the POSTed password, applying
+            // the hash against the pass / salt, if there is a match we
+            // found the user
+            hash(
```

Comment:
get user and authenticate user should be a separate methods

Reply:
Can you explane please? Need to separate part of authenticate method into another function or  you mean another API to get user information?

Reply:
yes, split them at least, please

Reply:
done